src/Microsoft.Repl.Tests/ShellTests.cs

```diff
@@ -0,0 +1,191 @@
+            Assert.Equal(inputBufferTextAfterKeyPress, shell.ShellState.InputManager.GetCurrentBuffer());
+        }
+
+        [Fact]
+        public void RunAsync_WithInsertKeyPress_FlipsIsOverwriteModeInInputManager()
+        {
+            Shell shell = CreateShell(ConsoleKey.Insert,
+                0,
+                null,
+                null,
+                out CancellationTokenSource cancellationTokenSource);
+
+            // Verify IsOverwriteMode flag in input manager is set to true before Insert key press event
+            Assert.False(shell.ShellState.InputManager.IsOverwriteMode);
+
+            shell.RunAsync(cancellationTokenSource.Token);
+
+            // Verify IsOverwriteMode flag in input manager is set to false after Insert key press event
+            Assert.True(shell.ShellState.InputManager.IsOverwriteMode);
+        }
+
+        [Fact]
+        public void RunAsync_WithUnhandledKeyPress_DoesNothing()
+        {
+            Shell shell = CreateShell(ConsoleKey.F1,
+                0,
+                null,
+                null,
+                out CancellationTokenSource cancellationTokenSource);
+
+            string inputBufferText = "get";
+
+            IShellState shellState = shell.ShellState;
+            shellState.InputManager.SetInput(shellState, inputBufferText);
+
+            // Verify the input buffer contents before F1 key press event
+            Assert.Equal(inputBufferText, shell.ShellState.InputManager.GetCurrentBuffer());
+
+            shell.RunAsync(cancellationTokenSource.Token);
+
+            // Verify the input buffer contents after F1 key press event
+            Assert.Equal(inputBufferText, shell.ShellState.InputManager.GetCurrentBuffer());
+        }
+
+        private Shell CreateShell(ConsoleKey consoleKey, int caretPosition, string previousCommand, string nextCommand, out CancellationTokenSource cancellationTokenSource)
+        {
+            HttpState httpState = new HttpState();
+            var defaultCommandDispatcher = DefaultCommandDispatcher.Create(httpState.GetPrompt, httpState);
+
+            Mock<IConsoleManager> mockConsoleManager = new Mock<IConsoleManager>();
+            cancellationTokenSource = new CancellationTokenSource();
+            CancellationToken cancellationToken = cancellationTokenSource.Token;
+            ConsoleKeyInfo consoleKeyInfo = new ConsoleKeyInfo('\0', consoleKey, false, false, false);
+            mockConsoleManager.SetupSequence(s => s.ReadKey(cancellationToken))
+                        .Returns(consoleKeyInfo);
+            mockConsoleManager.Setup(s => s.CaretPosition)
+                        .Returns(caretPosition);
+
+            Mock<ICommandHistory> mockCommandHistory = new Mock<ICommandHistory>();
+            mockCommandHistory.Setup(s => s.GetPreviousCommand())
+                        .Returns(previousCommand);
+            mockCommandHistory.Setup(s => s.GetNextCommand())
+            .Returns(nextCommand);
```

Comment:
nit: Indentation (and consistency of indentation) of the .Returns calls.

Reply:
Fixed here: https://github.com/aspnet/HttpRepl/commit/fb71f0807538c6980ebc937163f89cf70b330f31

Thanks!